library/src/updater.rs

```diff
@@ -138,6 +138,15 @@ pub fn init(
         NetworkHooks::default(),
     );
 
+    let _ = with_config(|config| {
+        UpdaterState::load_or_new_on_error(
+            &config.storage_dir,
+            &config.release_version,
+            config.patch_public_key.as_deref(),
+        )
+        .on_init()
+    });
+
```

Comment:
Currently set_config guards against being called twice, but this line does not.  So in the case of FCM (or similar) where there are multiple Flutter Engine instances being inited in parallel across multiple threads, we will end up calling on_init() twice, even though you probably don't want to.

We could make set_config return a bool or some other way to indicate failure and bail out.  set_config could return a Result<> and and Err in the case of already being init?  That would prevent us from reaching here.

Reply:
Fixed